controllers/components/dashboard/dashboard_controller.go

```diff
@@ -57,10 +57,16 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 		// By default, a predicated for changed generation is added by the Owns()
 		// method, however for deployments, we also need to retrieve status info
 		// hence we need a dedicated predicate to react to replicas status change
-		Owns(&appsv1.Deployment{}, builder.WithPredicates(resources.NewDeploymentPredicate())).
+		Owns(&appsv1.Deployment{}, reconciler.WithPredicates(resources.NewDeploymentPredicate())).
 		// operands - openshift
 		Owns(&routev1.Route{}).
 		Owns(&consolev1.ConsoleLink{}).
+		// Those APIs are provided by the component itself hence they should
+		// be watched dynamically
+		OwnsGVK(gvk.AcceleratorProfile, reconciler.Dynamic()).
+		OwnsGVK(gvk.OdhApplication, reconciler.Dynamic()).
+		OwnsGVK(gvk.OdhDocument, reconciler.Dynamic()).
+		OwnsGVK(gvk.OdhQuickStart, reconciler.Dynamic()).
```

Comment:
Should we watch `OdhDashboardConfig` for just for delete events?

Reply:
good point, I'll add it

Reply:
done